src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -257,6 +255,8 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $opt
 					return;
 				}
 			}
+
+			$this->isList = TrinaryLogic::createNo();
```

Comment:
This fix is needed or https://github.com/phpstan/phpstan-src/blob/2e2757395b0c3bd9a845e265f62803008b751bbe/tests/PHPStan/Rules/Properties/data/bug-3339.php#L4 will fail.

PHPStan was considering that
```
$this->tuple = [true, true, true];

		for ($i = 0; $i < 3; ++$i)
		{
			$this->tuple[$i] = false;
		}

// Here, `$this->tuple` is array{bool, bool, bool} but is not considered as a list
```

I added a non regression test [array-is-list-offset.php](https://github.com/phpstan/phpstan-src/pull/3381/files#diff-27c8d6ba604b96b0c4c1daab0a1520d5dee644d212dd58c16e327ca8b807b2de) which fails without this fix.

The idea is to not touch the `list` property if we're in the case
```
if ($match) {
					$this->valueTypes = $valueTypes;
					return;
				}
```
were all the constantScalar keys are already matching.